daemon/list.go

```diff
@@ -137,6 +138,14 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			Names: names[container.ID],
 		}
 		newC.Image = container.Config.Image
+		if container.Config.Image != "" {
+			img, err := daemon.repositories.LookupImage(container.Config.Image)
```

Comment:
This change is to return the full image ID which length is 64 if the `container.Config.Image` is a image ID and if the `container.Config.Image` is a image name, then just return the image name. Why we do this is `docker ps` has a option `--no-trunc`, with this option `docker ps` should show the full image ID.